cardano-api/src/Cardano/Api/IPC/Monad.hs

```diff
@@ -21,40 +22,113 @@ import Data.Ord
 import Shelley.Spec.Ledger.Scripts ()
 import System.IO
 
+import qualified Ouroboros.Network.Protocol.ChainSync.Client as Net.Sync
 import qualified Ouroboros.Network.Protocol.LocalStateQuery.Client as Net.Query
 import qualified Ouroboros.Network.Protocol.LocalStateQuery.Type as Net.Query
 
 -- | Monadic type for constructing local state queries.
 --
 -- Use 'sendMsgQuery' in a do block to construct queries of this type and convert
--- the expression to a 'Net.Query.LocalStateQueryClient' with 'setupLocalStateQueryScript'.
-newtype LocalStateQueryScript block point query r m a = LocalStateQueryScript
-  { runLocalStateQueryScript :: ContT (Net.Query.ClientStAcquired block point query m r) m a
+-- the expression to a 'Net.Query.LocalStateQueryClient' with 'setupLocalStateQueryExpr'.
+newtype LocalStateQueryExpr block point query r m a = LocalStateQueryExpr
+  { runLocalStateQueryExpr :: ContT (Net.Query.ClientStAcquired block point query m r) m a
   } deriving (Functor, Applicative, Monad, MonadIO)
 
 -- | Use 'sendMsgQuery' in a do block to construct monadic local state queries.
-sendMsgQuery :: Monad m => query a -> LocalStateQueryScript block point query r m a
-sendMsgQuery q = LocalStateQueryScript . ContT $ \f -> pure $
+sendMsgQuery :: Monad m => query a -> LocalStateQueryExpr block point query r m a
```

Comment:
Typically we will want to use the IO monad right? If that's the case, lets concretize that and make things less generic.

Reply:
Yep we typically do.